internal/backend/auth/authhttpmiddleware/authmiddleware.go

```diff
@@ -34,19 +36,25 @@ type Deps struct {
 	Tokens   authtokens.Tokens  `optional:"true"`
 }
 
-func newTokensMiddleware(next http.Handler, tokens authtokens.Tokens) http.HandlerFunc {
+func newTokensMiddleware(next http.Handler, deps Deps) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		ctx := r.Context()
-		ctx = akCtx.WithRequestOrginator(ctx, akCtx.User)
-
-		if user := authcontext.GetAuthnUser(ctx); !user.IsValid() {
-			if auth := r.Header.Get("Authorization"); auth != "" {
-				kind, payload, _ := strings.Cut(auth, " ")
+		ctx := akCtx.WithRequestOrginator(r.Context(), akCtx.User)
+		user := authcontext.GetAuthnUser(ctx)
+		authHdr := r.Header.Get("Authorization")
 
+		if deps.Cfg.AllowDefaultUser {
+			if user.IsValid() || authHdr != "" {
```

Comment:
Is it possible that the user you got from context is the default user ?
If so, this check would not allow the default user to work

Reply:
in my view - default user is not a user that could have token or could be provided externally.
We may consider change it of coz (for example create default credentials file thus providing default user in cli), but meanwhile I see no reason - and backend is handling lack of the user by attaching default user to the request